src/components/learner-credit-management/data/hooks/useBudgetDetailTabs.jsx

```diff
@@ -59,7 +57,7 @@ export const useBudgetDetailTabs = ({
           }
           className={TAB_CLASS_NAME}
         >
-          {activeTabKey === BUDGET_DETAIL_CATALOG_TAB && (
+          {(activeTabKey === BUDGET_DETAIL_CATALOG_TAB && !isRetired) && (
```

Comment:
I would recommend continuing rendering the `CatalogTabElement` component when `activeTabKey === BUDGET_DETAIL_CATALOG_TAB`. 

`activeTabKey` just never will be equal to `BUDGET_DETAIL_CATALOG_TAB` due to the `disabled={isRetired}` above. The extra check for `!isRetired` isn't really needed at that point, as the component never actually gets rendered until the "Catalog" tab is selected (but can't be due to disabled state).